### ISSUE TITLE
NovaExceptionHandler appears to be handling non-Nova exceptions

### ISSUE THREAD

**Issue by dwightwatson** [laravel/nova-issues#3867]
- Laravel Version: 9.7.0
- Nova Version: 4.0.5
- PHP Version: 8.1.4

### Description:

I'm opening this issue, a continuation of #3851. In my application I had upgraded from Nova 3 to Nova 4.

I've copied the stacktrace from that issue as an example:

```
View [app] not found. {"userId":7,"exception":"[object] (InvalidArgumentException(code: 0): View [app] not found. at /Users/cajunmba/LaravelProjects/signs/vendor/laravel/framework/src/Illuminate/View/FileViewFinder.php:137)
[stacktrace]
#0 /Users/cajunmba/LaravelProjects/signs/vendor/laravel/framework/src/Illuminate/View/FileViewFinder.php(79): Illuminate\\View\\FileViewFinder->findInPaths('app', Array)
#1 /Users/cajunmba/LaravelProjects/signs/vendor/laravel/framework/src/Illuminate/View/Factory.php(138): Illuminate\\View\\FileViewFinder->find('app')
#2 /Users/cajunmba/LaravelProjects/signs/vendor/laravel/framework/src/Illuminate/Routing/ResponseFactory.php(85): Illuminate\\View\\Factory->make('app', Array)
#3 /Users/cajunmba/LaravelProjects/signs/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(261): Illuminate\\Routing\\ResponseFactory->view('app', Array)
#4 /Users/cajunmba/LaravelProjects/signs/vendor/inertiajs/inertia-laravel/src/Response.php(113): Illuminate\\Support\\Facades\\Facade::__callStatic('view', Array)
#5 /Users/cajunmba/LaravelProjects/signs/vendor/laravel/nova/src/Exceptions/NovaExceptionHandler.php(88): Inertia\\Response->toResponse(Object(Illuminate\\Http\\Request))
#6 /Users/cajunmba/LaravelProjects/signs/vendor/laravel/nova/src/Exceptions/NovaExceptionHandler.php(68): Laravel\\Nova\\Exceptions\\NovaExceptionHandler->renderInertiaException(Object(Illuminate\\Http\\Request), Object(InvalidArgumentException))
#7 /Users/cajunmba/LaravelProjects/signs/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(51): Laravel\\Nova\\Exceptions\\NovaExceptionHandler->render(Object(Illuminate\\Http\\Request), Object(InvalidArgumentException))
#8 /Users/cajunmba/LaravelProjects/signs/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(172): Illuminate\\Routing\\Pipeline->handleException(Object(Illuminate\\Http\\Request), Object(InvalidArgumentException))
#9 /Users/cajunmba/LaravelProjects/signs/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/PreventRequestsDuringMaintenance.php(86): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#10 /Users/cajunmba/LaravelProjects/signs/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\PreventRequestsDuringMaintenance->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#11 /Users/cajunmba/LaravelProjects/signs/vendor/fruitcake/laravel-cors/src/HandleCors.php(38): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#12 /Users/cajunmba/LaravelProjects/signs/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Fruitcake\\Cors\\HandleCors->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#13 /Users/cajunmba/LaravelProjects/signs/vendor/laravel/framework/src/Illuminate/Http/Middleware/TrustProxies.php(39): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#14 /Users/cajunmba/LaravelProjects/signs/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Http\\Middleware\\TrustProxies->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#15 /Users/cajunmba/LaravelProjects/signs/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(103): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#16 /Users/cajunmba/LaravelProjects/signs/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(142): Illuminate\\Pipeline\\Pipeline->then(Object(Closure))
#17 /Users/cajunmba/LaravelProjects/signs/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(111): Illuminate\\Foundation\\Http\\Kernel->sendRequestThroughRouter(Object(Illuminate\\Http\\Request))
#18 /Users/cajunmba/LaravelProjects/signs/public/index.php(52): Illuminate\\Foundation\\Http\\Kernel->handle(Object(Illuminate\\Http\\Request))
#19 /Users/cajunmba/.composer/vendor/laravel/valet/server.php(219): require('/Users/cajunmba...')
#20 {main}
"}
```

After deploying to production I started getting the error `View [app] not found.` It appears as though this is caused by the `NovaExceptionHandler` trying to handle an exception with an Inertia response, and attempting to load an `app` view not in the main app.

I noticed this affected two URL patterns before I rolled back to Nova 3:

* `/forums/*` are not current URLs - they are an older pattern still indexed by search engines that should 404.
* `/livewire/*` are current Livewire URLs, but it looks like the actual exception is being gobbled up so I can't see what the issue is.

<img width="1255" alt="image" src="https://user-images.githubusercontent.com/1100408/162105487-b1c361c1-ec09-4b1e-8f84-9376b5c45f18.png">

It's almost like Nova is taking all routes I haven't defined (like these non-existing `/forums/*` and the Livewire defined `/livewire/*`) and attempting to serve those as if they were Nova requests?

**Comment by crynobone** [comment]
Unable to reproduce, see https://github.com/nova-issues/issue-3867/commit/7ec7cb93056e9383fd50c05e780d550c3da797ea

https://user-images.githubusercontent.com/172966/162110466-4bf689fa-6309-4f07-9aca-ca0f886fcbc8.mp4

**Comment by dwightwatson** [comment]
Cool, I'll keep diving to see if I can isolate the issue. So far I've only spotted the issue in production (on Vapor/Octane, not sure if that's a contributing factor).

**Comment by dwightwatson** [comment]
Alright - I think I've replicated it here. It looks like it is related to Octane - perhaps once it loads a Nova request it then treats following requests as Nova requests.

Here's the example repo: https://github.com/dwightwatson/nova-issue

1. Pull down the repo and get the database and user up and running.

```sh
touch database/database.sqlite
php artisan migrate
```

```php
(new App\Models\User)->forceFill(['name' => 'Dwight', 'email' => 'dwight@example.com', 'password' => bcrypt('password')])->save()
```

2. Start Octane and login

```sh
php artisan octane:start
```

Visit `https://localhost:8000/nova` and login using the new account. 

3. Visit a non-Nova URL in a new window

Click through Nova a bit, and then visit a non-Nova URL, like:  `http://localhost:8000/forums/showthread.php?p=3000`.

Sometimes it will 404 (as expected) but other times it crashes the Octane server. The first request was a 404, and then when I refreshed the same page I got the worker stopped notification.

```
  200    GET /nova-api/users .................................................................................................................. 199.88 ms
  404    GET /forums/showthread.php ............................................................................................................. 2.53 ms
worker stopped, and will be restarted
worker stopped
```

If I continue refreshing the page eventually goes back to a 404. However once I play around with Nova again I can replicate the crash.

---

Let me know if any of this doesn't make sense, or if you aren't able to replicate this on your machine. 

I'm using RoadRunner on my local machine, but in production I'm using Swoole on Vapor so it seems like it's a general Octane issue. I can't replicate this issue using Valet locally.
